File: src/api/MoRS.ManagementSystem.API/DTOs/Room/RoomResponse.cs
namespace MoRS.ManagementSystem.API.DTOs.Room;

public class RoomResponse
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Type { get; set; }
    public required string Color { get; set; }
    public bool IsActive { get; set; }
}

File: src/desktop/lib/models/room/room_response.dart
class RoomResponse {
  final int id;
  final String name;
  final String type;
  final String color;
  final bool isActive;

  RoomResponse({
    required this.id,
    required this.name,
    required this.type,
    required this.color,
    required this.isActive,
  });

  factory RoomResponse.fromJson(Map<String, dynamic> json) {
    return RoomResponse(
      id: json['id'],
      name: json['name'],
      type: json['type'],
      color: json['color'],
      isActive: json['isActive'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'color': color,
      'isActive': isActive,
    };
  }
}
